src/com/owncloud/android/ui/activity/UploadFilesActivity.java

```diff
@@ -376,26 +454,36 @@ protected Boolean doInBackground(Void... params) {
          */
         @Override
         protected void onPostExecute(Boolean result) {
-            mCurrentDialog.dismiss();
-            mCurrentDialog = null;
-            
+            if(requestCode == FileDisplayActivity.REQUEST_CODE__SELECT_FILES_FROM_FILE_SYSTEM) {
+                mCurrentDialog.dismiss();
+                mCurrentDialog = null;
+            }
             if (result) {
                 // return the list of selected files (success)
                 Intent data = new Intent();
-                data.putExtra(EXTRA_CHOSEN_FILES, mFileListFragment.getCheckedFilePaths());
+                if(requestCode == FileDisplayActivity.REQUEST_CODE__UPLOAD_FROM_CAMERA){
+                    data.putExtra(EXTRA_CHOSEN_FILES,new String[]{ capturedImagePath});
+                } else {
+                    data.putExtra(EXTRA_CHOSEN_FILES, mFileListFragment.getCheckedFilePaths());
+                }
 
                 SharedPreferences.Editor appPreferencesEditor = PreferenceManager
                         .getDefaultSharedPreferences(getApplicationContext()).edit();
 
-
-                if (mRadioBtnMoveFiles.isChecked()){
+                if(requestCode == FileDisplayActivity.REQUEST_CODE__UPLOAD_FROM_CAMERA){
```

Comment:
You are using the condition `if(requestCode == FileDisplayActivity.REQUEST_CODE__UPLOAD_FROM_CAMERA){` twice here to set several params, is not possible to use just one and unify all the code?

Reply:
Yes I have done that